packages/application/src/requests.tsx

```diff
@@ -197,7 +257,13 @@ export class RequestHandler {
             },
             // handle 404 if the server is not found
             (reason: any) => {
-              return this.server404(requestPath);
+              if (response.status >= 200 && response.status < 300) {
+                resolve();
+              } else if (response.status == 404) {
+                return this.server404(requestPath);
+              } else {
+                return this.serverError(reason);
+              }
```

Comment:
I'm inclined to **only** handle success status `204` in this block since any other successes are bugs because it means they returned content.
```suggestion
            
             if (response.status == 404) {
                return this.server404(requestPath);
             } else if (response.status == 204) {
                resolve(); 
             } else {
                return this.serverError(reason);
             }
```

Reply:
is 204 the only success case with no content, or the only one we support currently? If it's the former then I agree with this, but if it's the later then I would argue that returning nothing and letting the services that call this function handle that "unexpected empty" error case is better.

Reply:
Yes - its the only success status _we_ return w/o content and I believe that's the case in general: https://restfulapi.net/http-status-codes/.   It looks like 304 also requires the response to be empty, but I think we could cross that bridge as necessary (I doubt we'll need to do ever deal with 304 status codes TBH).

Technically speaking, I think we should cap our "success range" to < 300 since 3xx codes are "redirection" and not "success" codes anyway.

Reply:
In that case I'm ok with this change. @marthacryan if you want to merge this when you get started in the morning.